### ISSUE TITLE
Android LoginActivity throws NullPointerException when started by Android OS

### ISSUE THREAD

**Issue by mikedawson** [UstadMobile/UstadMobile#3]
If the LoginActivity is started directly by the OS without being started by startUI instead there is no associated LoginViewAndroid object and a nullpointexception is thrown : 

```
java.lang.RuntimeException: Unable to start activity ComponentInfo{com.toughra.ustadmobile/com.ustadmobile.port.android.view.LoginActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'void com.ustadmobile.port.android.view.LoginViewAndroid.setLoginViewActivity(com.ustadmobile.port.android.view.LoginActivity)' on a null object reference
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2314)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2388)
	at android.app.ActivityThread.access$800(ActivityThread.java:148)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1292)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:135)
	at android.app.ActivityThread.main(ActivityThread.java:5312)
	at java.lang.reflect.Method.invoke(Native Method)
	at java.lang.reflect.Method.invoke(Method.java:372)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:901)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:696)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'void com.ustadmobile.port.android.view.LoginViewAndroid.setLoginViewActivity(com.ustadmobile.port.android.view.LoginActivity)' on a null object reference
	at com.ustadmobile.port.android.view.LoginActivity.onCreate(LoginActivity.java:60)
	at android.app.Activity.performCreate(Activity.java:5953)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1128)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2267)
	... 10 more
```

**Comment by mikedawson** [comment]
Issue fixed: added design pattern to allow for instances where the underlying system (e.g. Android) restores a view and the controller needs loaded.

**Comment by mikedawson** [comment]
onDestroy/onCreate still not behaving as expected.  If developer options are set to destroy the activity immediately loadable controller based activities still misbehave.  The objects themselves are not disposed of by the Android OS - appcompat takes care of restoring the UI and then it mistakenly throws another fragment on top of itself.

CatalogActivity does not track the last active fragment and is saving the wrong URL as the last one used.

**Comment by mikedawson** [comment]
Fixed with commit 8427c10  - onCreate and onDestroy work both when the activity is started from a view and when the activity is started by the Android OS .
